tracer/src/Datadog.Trace/Util/ProcessHelpers.cs

```diff
@@ -174,9 +173,18 @@ private static class CurrentProcess
         {
             internal static readonly Process Instance;
```

Comment:
Given this is literally only called by the runtime metrics, and we have to call `Refresh()` on it anyway, shall we just stop caching it here (so that a similar mistake isn't made later) and just call `Process.GetCurrentProcess();` again in the runtime metrics path?

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/4567/commits/bdc7c7f78cfa54a9d5c1d6082149475e9b615046